src/main/java/com/minecolonies/structures/client/TemplateBlockAccessTransformHandler.java

```diff
@@ -0,0 +1,40 @@
+package com.minecolonies.structures.client;
+
+import net.minecraft.world.gen.structure.template.Template;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public class TemplateBlockAccessTransformHandler
+{
+    private static TemplateBlockAccessTransformHandler ourInstance = new TemplateBlockAccessTransformHandler();
+
+    public static TemplateBlockAccessTransformHandler getInstance()
+    {
+        return ourInstance;
+    }
+
+    private Map<Predicate<Template.BlockInfo>, Function<Template.BlockInfo, Template.BlockInfo>> blockInfoTransformHandler = new HashMap<>();
+
+    private TemplateBlockAccessTransformHandler()
+    {
+    }
+
+    public void AddTransformHandler(@NotNull final Predicate<Template.BlockInfo> transformPredicate, @NotNull final Function<Template.BlockInfo, Template.BlockInfo> transformHandler)
+    {
+        blockInfoTransformHandler.put(transformPredicate, transformHandler);
+    }
+
+    public Template.BlockInfo Transform(@NotNull final Template.BlockInfo blockInfo)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00100)